smarts/core/lanepoints.py

```diff
@@ -881,7 +881,7 @@ def closest_lanepoints(
             k=maximum_count,
             filter_composites=True,
         )
-        return [l_lps[0].lp for l_lps in linked_lanepoints]
+        return [l_lps.lp for l_lps in linked_lanepoints[0]]
```

Comment:
What is the purpose of switching from row to column selection?

```
Before
+---+---+---+
|p1 |p2 |p3 |
+---+---+---+
| x | x | x |
+---+---+---+
|   |   |   |
+---+---+---+
After
+---+---+---+
|p1 |p2 |p3 |
+---+---+---+
| x |   |   |
+---+---+---+
| x |   |   |
+---+---+---+
```

Am I mistaken in thinking that now all the `lps` will be related to the first pose?

Reply:
Good catch. I didn't even realize that this takes in multiple poses. I'm a bit confused by the intention of this function, considering it seems to only ever be called with 1 pose. The original definition only took the first lanepoint of each set of 10 lanepoints for each pose. I think I may just change it to take a single pose and leave it as-is.